src/wp-includes/global-styles-and-settings.php

```diff
@@ -374,6 +364,45 @@ function ( $item ) {
 	}
 }
 
+/**
+ * Gets the block name from a given theme.json path.
+ *
+ * @since 6.3.0
+ * @access private
+ *
+ * @param array $path An array of keys describing the path to a property in theme.json.
+ * @return string Identified block name, or empty string if none found.
+ */
+function wp_get_block_name_from_theme_json_path( $path ) {
+	// Block name is expected to be the third item after 'styles' and 'blocks'.
+	if (
+		count( $path ) >= 3
+		&& 'styles' === $path[0]
+		&& 'blocks' === $path[1]
+		&& str_contains( $path[2], '/' )
+	) {
+		return $path[2];
+	}
+
+	// As fallback and for backward compatibility, allow any core block to be
+	// at any position.
+	$result = array_values(
+		array_filter(
+			$path,
+			static function ( $item ) {
+				if ( strpos( $item, 'core/' ) !== false ) {
```

Comment:
Is `str_contains()` available here?

Reply:
Given that this is only checking for the existence of `core/`, not whether it comes first, I think it would be safer to make it use `str_contains()` for zero risk. Note that while it looks like this code is new, it was just moved here from `wp_add_global_styles_for_blocks()`.